src/particles/spacecharge/Gauss3dPush.cpp

```diff
@@ -0,0 +1,187 @@
+            amrex::ParticleReal const bt2sqrt = std::sqrt(bt2);
+            amrex::ParticleReal const f2x = t2sqrt * bt2sqrt;
+            amrex::ParticleReal const f2y = t2 * t2sqrt * bt2sqrt;
+            amrex::ParticleReal const f2z = bt2 * t2sqrt * bt2sqrt;
+            amrex::ParticleReal const f1 = x * x * std::exp(-x * x * (xp2 + yp2 / t2 + zp2 / bt2) / 2_prt);
+
+            // Simpson's rule integration
+
+            // Ex
+            amrex::ParticleReal const fx = f1 / f2x;
+            if (i%2 == 0)
+                sum0ex += 2_prt * fx * h;
+            else
+                sum0ex += 4_prt * fx * h;
+
+            // Ey
+            amrex::ParticleReal const fy = f1 / f2y;
+            if (i%2 == 0)
+                sum0ey += 2_prt * fy * h;
+            else
+                sum0ey += 4_prt * fy * h;
+
+            // Ez
+            amrex::ParticleReal const fz = f1 / f2z;
+            if (i%2 == 0)
+                sum0ez += 2_prt * fz * h;
+            else
+                sum0ez += 4_prt * fz * h;
+        }
+
+        pintex = sum0ex / 3_prt;
+        pintey = sum0ey / 3_prt;
+        pintez = sum0ez / 3_prt;
+    }
+
+    void Gauss3dPush (
+        ImpactXParticleContainer & pc,
+        amrex::ParticleReal const slice_ds
+    )
+    {
+        BL_PROFILE("impactx::spacecharge::GatherAndPush");
+
+        using namespace amrex::literals;
+
+        amrex::ParticleReal const charge = pc.GetRefParticle().charge;
+
+        // get RMS sigma of beam
+        // Standard deviation of the particle positions (speed of light times time delay for t) (unit: meter)
+        std::unordered_map<std::string, amrex::ParticleReal> const rbc = diagnostics::reduced_beam_characteristics(pc);
+        amrex::ParticleReal const sigx = rbc.at("sig_x");
+        amrex::ParticleReal const sigy = rbc.at("sig_y");
+        amrex::ParticleReal const sigz = rbc.at("sig_t");
+        amrex::ParticleReal const bchchg = rbc.at("charge_C");
+
+        // physical constants and reference quantities
+        amrex::ParticleReal const c0_SI = 2.99792458e8;  // TODO move out
+        amrex::ParticleReal const mc_SI = pc.GetRefParticle().mass * c0_SI;
+        amrex::ParticleReal const pz_ref_SI = pc.GetRefParticle().beta_gamma() * mc_SI;
+        amrex::ParticleReal const gamma = pc.GetRefParticle().gamma();
+        amrex::ParticleReal const inv_gamma2 = 1.0_prt / (gamma * gamma);
+        amrex::ParticleReal const rfpiepslon = c0_SI*c0_SI*1.0e-7;
+
+        amrex::ParticleReal const dt = slice_ds / pc.GetRefParticle().beta() / c0_SI;
+
+        // group together constants for the momentum
+        using ablastr::constant::math::pi;
+        amrex::ParticleReal const asp = sigx/sigy;
+        amrex::ParticleReal const push_consts = dt * charge * inv_gamma2 / pz_ref_SI
+            * 2_prt * asp * (sigx * sigx * sigz * std::sqrt(2_prt * pi))
+            * bchchg * rfpiepslon;
```

Comment:
These need a separate constant per x,y,z, which I accidentally removed when refactoring.

Fixed in #1168